paimon-core/src/main/java/org/apache/paimon/iceberg/manifest/IcebergConversions.java

```diff
@@ -74,8 +79,32 @@ public static ByteBuffer toByteBuffer(DataType type, Object value) {
             case DECIMAL:
                 Decimal decimal = (Decimal) value;
                 return ByteBuffer.wrap((decimal.toUnscaledBytes()));
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+                TimestampType timestampType = (TimestampType) type;
+                precision = timestampType.getPrecision();
+                timestamp = (Timestamp) value;
+                return convertTimestampWithPrecisionToBuffer(timestamp, precision);
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                LocalZonedTimestampType localTimestampType = (LocalZonedTimestampType) type;
+                precision = localTimestampType.getPrecision();
+                timestamp = (Timestamp) value;
+                return convertTimestampWithPrecisionToBuffer(timestamp, precision);
+            case TIME_WITHOUT_TIME_ZONE:
+                Long time = ((Integer) value).longValue();
+                return ByteBuffer.allocate(8).order(ByteOrder.LITTLE_ENDIAN).putLong(0, time);
             default:
                 throw new UnsupportedOperationException("Cannot serialize type: " + type);
         }
     }
+
+    private static ByteBuffer convertTimestampWithPrecisionToBuffer(
+            Timestamp timestamp, int precision) {
+        long timestampValue;
+        if (precision <= 6) {
+            timestampValue = timestamp.getMillisecond();
```

Comment:
According to Iceberg's document: Stores **microseconds** from 1970-01-01 00:00:00.000000 in an 8-byte little-endian long.

`timestamp.getMillisecond()` only gives you millisecond. Also don't forget there might still be values in `getNanoOfMillisecond`.